src/espidf.rs

```diff
@@ -102,6 +102,206 @@ impl Tools {
+            .as_array()
+            .and_then(|path_array| path_array.iter().map(|path| path.as_str()).collect())
+            .unwrap_or_else(Vec::new);
+
+        log::debug!("export_path: {export_path:?}");
+
+        let version_cmd_args = tool_object["version_cmd"]
+            .as_array()
+            .unwrap()
+            .iter()
+            .map(|v| v.as_str().unwrap().to_string())
+            .collect::<Vec<String>>();
+
+        let version_regex = tool_object["version_regex"].as_str().unwrap().to_string();
+
+        let mut tool = Tool {
+            name: name.to_string(),
+            url: String::new(),
+            versions: String::new(),
+            sha256: String::new(),
+            size: 0,
+            install_dir: install_dir.clone(),
+            export_path: PathBuf::new(),
+            version_cmd_args,
+            version_regex,
+        };
+
+        // pick the recommended version out of a list of versions for the current os
+        tool_object
+            .get("versions")
+            .and_then(|versions| versions.as_array())
+            .unwrap()
+            .iter()
+            .filter(|version| {
+                // filter by version object where key "status" is "recommended"
+                let inner = version.as_object().unwrap();
+                inner.get("status").unwrap().as_str() == Some("recommended")
+            })
+            .for_each(|version| {
+                // only insert the version object if it contains the correct os key
+                let inner = version.as_object().unwrap();
+                if let Some(os_version) = inner.get(os_key) {
+                    if let Some(url) = os_version.get("url") { tool.url = url.as_str().unwrap().to_string(); }
+                    if let Some(sha256) = os_version.get("sha256") { tool.sha256 = sha256.as_str().unwrap().to_string(); }
+                    if let Some(size) = os_version.get("size") { tool.size = size.as_u64().unwrap(); }
+                    if let Some(name) = version.get("name") { tool.versions = name.as_str().unwrap().to_string(); }
+
+                    tool.export_path = PathBuf::new()
+                        .join("tools")
+                        .join(&tool.name)
+                        .join(&tool.versions);
+                    for p in export_path.iter() {
+                        tool.export_path = tool.export_path.join(p);
+                    }
+                }
+            });
+
+        tools.push(tool);
+    }
+    log::debug!("============================================================================");
+
+    Ok(tools)
+}
+
+// Maps the current os and architecture to the correct key in the tools.json file
+fn get_os_target_key() -> Option<&'static str> {
+    let os = std::env::consts::OS;
+    let arch = std::env::consts::ARCH;
+
+    match os {
+        "linux" => match arch {
+            "x86_64" => Some("linux-amd64"),
+            // TODO add and test arm variants
+            _ => None,
```

Comment:
Might want to at least add support for RPI?

Reply:
arm is kinda a wormhole on linux ;D for rpi 3 & 4 & 5 i think we need different targets and it depends on what OS are flashed onto the pi. E.g a stock "pi os" or some debian version. I think they are all "linux" on the os, but the second part is more unclear. Do you know working matching patterns for the "arch" variants for the different thumb and aarach variants on pi3 -pi5. IIrc they are all different.

Reply:
added linux arm variants the way i think its correct